feat(trajectory):  refactor velocity_smoother (jerk smoother function) to experimental trajectory

mraditya01 opened the pull request:
## Description

Converted velocity smoother package (analytical_jerk_constrained_smoother) from discrete `TrajectoryPoints` (vector-based) to continuous `Trajectory`. Will remove the discrete function by the end of this PR.

### Migration Status
| Function Name | Location/File | Status |
|:--------------|:--------------|:-------|
| `apply` | `jerk_filtered_smoother.cpp` | ✅ |
| `forwardJerkFilter` | `jerk_filtered_smoother.cpp` | ✅ |
| `backwardJerkFilter` | `jerk_filtered_smoother.cpp` | ✅ |
| `mergeFilteredTrajectory` | `jerk_filtered_smoother.cpp` | ✅ |
| `resampleTrajectory` | `jerk_filtered_smoother.cpp` | ❌ Not needed (continuous `Trajectory` API) |
| `apply` | `l2_pseudo_jerk_smoother.cpp` | ✅ |
| `resampleTrajectory` | `l2_pseudo_jerk_smoother.cpp` | ❌ Not needed|
| `apply` | `linf_pseudo_jerk_smoother.cpp` | ✅ |
| `resampleTrajectory` | `linf_pseudo_jerk_smoother.cpp` | ❌ Not needed |
| `applyPreProcess` | `smoother_base.cpp` | ✅ |
| `applyLateralAccelerationFilter` | `smoother_base.cpp` | ✅ |
| `applySteeringRateLimit` | `smoother_base.cpp` | ✅  |
| `VelocitySmoother` | `node.cpp` | ✅  |


</body></html>

## Related links

**Related PR:** https://github.com/autowarefoundation/autoware_core/pull/749, https://github.com/autowarefoundation/autoware_core/pull/762

## How was this PR tested?

[4289/4289 CommonScenario_basic:4
](https://evaluation.tier4.jp/evaluation/reports/4750c39e-2ce7-5e47-9b39-34315e5e080f?project_id=autoware_dev)

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

mraditya01 committed:
initial commit, feat: refactor l2_pseudo and linf_pseudo

mraditya01 committed:
refactor more function, checked build

mraditya01 committed:
feat: refactor smoother_base.cpp build and tested

mraditya01 committed:
fix: fixed continuous applySteeringRateLimit function

mraditya01 committed:
feat: update node to use continuous trajectory during smoothing

mraditya01 committed:
Merge branch 'main' into feat/additional_smoother_refactor

mraditya01 committed:
missing header for new class

mraditya01 committed:
update node to the newest

pre-commit-ci[bot] committed:
style(pre-commit): autofix

mraditya01 committed:
fixed jerk calculation and added shield

mraditya01 committed:
revert test deletion

mraditya01 committed:
fix cmake

mraditya01 committed:
fix: modify jerk position in continuous

mraditya01 committed:
revert back and avoid restore()

mraditya01 committed:
revert back and avoid restore()

mraditya01 committed:
revert restore()

soblin commented on planning/autoware_velocity_smoother/src/trajectory_utils.cpp:
```diff
@@ -230,6 +267,19 @@ void applyMaximumVelocityLimit(
   }
 }
 
+void applyMaximumVelocityLimit(
```
Seems like it's not necessary anymore

sarun-hub replied:
PR #749 already removed the changed in the commit, and it was merged to this branch in commit [bdf84ee](https://github.com/autowarefoundation/autoware_core/pull/763/commits/bdf84eef303d576dfb4873b39278c78bbb59e935).

soblin commented on planning/autoware_velocity_smoother/src/trajectory_utils.cpp:
```diff
@@ -220,6 +236,27 @@ std::vector<double> calcTrajectoryCurvatureFrom3Points(
   return k_arr;
 }
 
+std::vector<double> calcTrajectoryCurvatureFrom3Points(
```
Seems like it's not necessary anymore

sarun-hub replied:
PR #749 already removed the changed in the commit, and it was merged to this branch in commit [bdf84ee](https://github.com/autowarefoundation/autoware_core/pull/763/commits/bdf84eef303d576dfb4873b39278c78bbb59e935).

mraditya01 committed:
Merge branch 'main' into feat/additional_smoother_refactor

sarun-hub committed:
Merge branch 'main' into feat/additional_smoother_refactor

sarun-hub committed:
add and remove tests in CMakeLists and adjust test

Signed-off-by: Sarun Mukdapitak <sarun.mukda@gmail.com>

sarun-hub committed:
fix cppcheck-differential

1) reorder of condition to allow size check before value check
2) remove knownCondition, there are condition ensuring N >=2, thus remove checking N>0

Signed-off-by: Sarun Mukdapitak <sarun.mukda@gmail.com>

sarun-hub committed:
fix clang-tidy-error

1) bugprone-narrowing-conversions
2) clang-diagnostic-error
3) clang-diagnostic-inconsistent-missing-override

Signed-off-by: Sarun Mukdapitak <sarun.mukda@gmail.com>

sarun-hub committed:
Merge branch 'main' into feat/additional_smoother_refactor

sarun-hub committed:
fix spell-check, use full word of velocity

Signed-off-by: Sarun Mukdapitak <sarun.mukda@gmail.com>

sarun-hub committed:
fix type conversion (clangtidy)

Signed-off-by: Sarun Mukdapitak <sarun.mukda@gmail.com>

sarun-hub committed:
Merge branch 'main' into feat/additional_smoother_refactor

sarun-hub committed:
Merge branch 'main' into feat/additional_smoother_refactor

sarun-hub committed:
Merge branch 'main' into feat/additional_smoother_refactor
